src/helpers/snippet.ts

```diff
@@ -1,16 +1,26 @@
+import { Hit } from '../types';
 import { getPropertyByPath } from '../lib/utils';
 import { TAG_REPLACEMENT } from '../lib/escape-highlight';
 import { component } from '../lib/suit';
 
+export type SnippetOptions = {
+  attribute: string;
+  highlightedTagName?: string;
+  hit: Partial<Hit>;
+};
+
 const suit = component('Snippet');
 
-export default function snippet({
-  attribute,
-  highlightedTagName = 'mark',
-  hit,
-} = {}) {
+export default function snippet(
+  {
+    attribute,
+    highlightedTagName = 'mark',
+    hit,
+  }: SnippetOptions = {} as SnippetOptions
```

Comment:
if it is `= {}` then doing `as SnippetOptions` can cause undefined errors. In fact I think it probably should be removed like this:

```suggestion
  }: SnippetOptions
```

Reply:
Good point – will remove this default value.